Assets/Scripts/Controllers/CharacterController.cs

```diff
@@ -5,15 +5,24 @@
 
 public class CharacterController : MonoBehaviour
 {
+    [SerializeField] private GameObject _weaponProjectile;
     private Rigidbody2D _rb;
     private Vector2 _movement;
+    private int _fireRate = 0;
```

Comment:
Be sure to comment any fields added to class members to ensure that other members can understand your code as quickly as possible:
`_fireRate = 0; // Number of frames between two projectiles being spawned.`
`_projectileSpawnModifier; // Offset to determine spawn position of projectiles.`

In general, prefer verbosity over assuming people will know what you mean!